### PR TITLE
[Fixes #7020] Remove stale templates and hooksets implementations fro…

### LINKED ISSUE #7020: Remove stale templates and hooksets implementations from "client" module
The client module still ships old and unused code and templates (e.g.Leaflet preview), and it implements hooksets for Leaflet and React that shouldn't stay inside the core.
Let's prune this stale code and just leave the generic hooks, that will be revised in the context of the client architecture refactoring.

OL templates must be kept for the moment since they're used inside the layer and map metadata wizard.

### PR DESCRIPTION
**Pull request by afabiani**
…m "client" module

Part of: GNIP 81 - GeoNode Core Cleanup #6917
References: #7020

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Review by giohappy — DISMISSED** [review]

**Review by t-book — COMMENTED** [review]
@afabiani I guess we can/should remove leaflet from static deps as well? https://github.com/GeoNode/geonode/blob/master/geonode/static/static_dependencies.json
Do not know of any part where it is still used, do you?

**Comment by afabiani** [comment]
> @afabiani I guess we can/should remove leaflet from static deps as well? https://github.com/GeoNode/geonode/blob/master/geonode/static/static_dependencies.json
> Do not know of any part where it is still used, do you?

Will do, thanks!

**Comment by afabiani** [comment]
@t-book pushed a commit to get rid of leaflet static stuff too

**Comment by t-book** [comment]
@afabiani awesome! spring cleaning!

**Comment by t-book** [comment]
@afabiani looks good to me. I saw you also build the frontend again. Have you test it, or should I check out and test it again?

**Comment by afabiani** [comment]
@t-book I tested it, but a second look would be very helpful!

**Review by t-book — DISMISSED** [review]

**Comment by t-book** [comment]
@afabiani checked, no problems no console errors. Speaking geo, a full extent of thanks ;))

File: geonode/local_settings.py.geoserver.sample [hunk 1]
```diff
@@ -288,89 +288,6 @@ GEONODE_CLIENT_LAYER_PREVIEW_LIBRARY = os.getenv('GEONODE_CLIENT_LAYER_PREVIEW_L
...
-            'leaflet-plugins': {
-                'js': 'lib/js/leaflet-plugins.min.js',
-                'css': 'lib/css/leaflet-plugins.min.css',
-                'auto-include': True,
-            }
-        }
-
 """
 To enable the MapStore2 REACT based Client:
 1. pip install pip install django-geonode-mapstore-client==1.0
```
Comment (gannebamm) [inline geonode/local_settings.py.geoserver.sample:293]: Is this deprecated?
`1. pip install pip install django-geonode-mapstore-client==1.0`
 Nonetheless, it is not part of this PR

File: geonode/layers/views.py [hunk 2]
```diff
@@ -1502,28 +1490,23 @@ def layer_thumbnail(request, layername):
         raise Http404(_("Not found"))
 
     try:
+        # try:
```
Comment (gannebamm) [inline geonode/layers/views.py:1493]: Why are those commented parts of the code in this PR?

**Review by gannebamm — COMMENTED** [review]
Check my two comments, apart from that the code seems fine. Functionality was tested by @t-book therefore I have not tested it again and did not deploy this code.

File: geonode/local_settings.py.geoserver.sample (hunk 1 above)
Comment (afabiani) [reply geonode/local_settings.py.geoserver.sample:293]: Yep, that version is old. That should be removed from the comments.

File: geonode/layers/views.py (hunk 2 above)
Comment (afabiani) [reply geonode/layers/views.py:1493]: Custom if-else statements used by leaflet only. Bad practice indeed. The idea is to remove all those if-else hardcoded into the code.

**Review by gannebamm — APPROVED** [review]
